testutils: roachtest errgroup import linter

herkolategan opened the pull request:
Previously, it was possible to import `golang.org/x/sync/errgroup` in roachtests. This change blocks the import via a linter that advises the usage of the Task API provided by `roachtestutils`. All usages of error groups have been removed and replaced with the Task API, See: #138372 for more details.

Epic: None
Release note: None

herkolategan committed:
testutils: roachtest errgroup import linter

Previously, it was possible to import `golang.org/x/sync/errgroup` in
roachtests. This change blocks the import via a linter that advises the usage of
the Task API provided by `roachtestutils`. All usages of error groups have been
removed and replaced with the Task API, See: #138372 for more details.

Epic: None
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139488)
<!-- Reviewable:end -->

herkolategan committed:
roachtest: replace ctxgroup with Task API

Previously, a few roachtest are still using `ctxgroup`. In an effort to have
roachtest manage all goroutines this has been replaced with the roachtest Task
API as well.

For simplicity the general logic has been kept the same, although some of these
tests do not technically need the use of an error group, and could use a regular
group that will fatal on a failure.

Informs: #118214

Epic: None
Release Note: None

herkolategan commented:
TFTR!
bors r=golgeek,Darrylwong
